### PR TITLE
refactor/cleanup

### PR DESCRIPTION
**Pull request by dtrifiro**
- rename client/configuration classes
- rename test modules
- add example
- misc bug fixes

### CODE REVIEW

File: examples/example.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from caikit_nlp_client import GrpcClient, GrpcConfig, HttpClient, HttpConfig
+
+text = "What is the boilign point of Nitrogen?"
+
+# Using the http client
+client = HttpClient(HttpConfig(host="localhost", port=8080, tls=False))
```
Comment (dtrifiro) [inline examples/example.py:6]: Still unsure whether it's worth it to have `HttpConfig`, `GrpcConfig` classes to just pass a few values to the init functions

File: src/caikit_nlp_client/__init__.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# noqa: F401
+from .grpc_client_introspection import GrpcClient, GrpcConfig
+from .http_client import HttpClient, HttpConfig
+
+__all__ = ["GrpcConfig", "GrpcClient", "HttpClient", "HttpConfig"]
```
Comment (dtrifiro) [inline src/caikit_nlp_client/__init__.py:5]: running `from caikit_nlp_client import *` will bring in all of these

File: src/caikit_nlp_client/grpc_channel.py [hunk 3]
```diff
@@ -1,44 +0,0 @@
-import logging
```
Comment (dtrifiro) [inline src/caikit_nlp_client/grpc_channel.py:1]: Moved this to `grpc_client.py`

File: src/caikit_nlp_client/grpc_client_introspection.py [hunk 4]
```diff
@@ -10,14 +12,61 @@
...
+
+class GrpcClient:
     """GRPC client for a caikit nlp runtime server
 
     Args:
         channel (grpc.Channel): a connected GRPC channel for use of making the calls.
     """
 
-    def __init__(self, channel: grpc.Channel) -> None:
+    def __init__(self, config: GrpcConfig) -> None:
```
Comment (dtrifiro) [inline src/caikit_nlp_client/grpc_client_introspection.py:59]: Changed the API so that it takes a config instead of a channel: the user will not care about the underlying channel

File: examples/example.py (hunk 1 above)
Comment (z103cb) [reply examples/example.py:6]: The config classes are a wee big javaesque. They can be removed, but the side effect is that there will be a lot of values that are optional. I suppose the idimatic python is to use `kwargs``.

File: src/caikit_nlp_client/__init__.py (hunk 2 above)
Comment (z103cb) [reply src/caikit_nlp_client/__init__.py:5]: cool

File: src/caikit_nlp_client/grpc_client_introspection.py (hunk 4 above)
Comment (z103cb) [reply src/caikit_nlp_client/grpc_client_introspection.py:59]: channels need to be closed, having them exposed puts the onus of the client to manage the channels.

File: src/caikit_nlp_client/grpc_client_introspection.py [hunk 5]
```diff
@@ -10,14 +12,61 @@
...
+        """Client class for a Caikit NLP grpc server
+
+        >>> client = GrpcClient(GrpcConfig(host="localhost", port="8085"))
+        >>> generated_text = client.generate_text_stream(
+                "flan-t5-small-caikit",
+                "What is the boiling point of Nitrogen?"
+            )
+        """
+
+        channel = make_channel(config)
```
Comment (z103cb) [inline src/caikit_nlp_client/grpc_client_introspection.py]: Channels are sockets... i think we are leaking them with this implementation.

File: tests/test_api.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
+def test_module():
```
Comment (z103cb) [inline tests/test_api.py:1]: Why is this necessary ?

**Review by z103cb — COMMENTED** [review]
I think that this is the huge step in the right direction. It is not clear to me what is the "pythonic" way of managing the lifecycle of channels. We should probably  have htat in the code

File: examples/example.py (hunk 1 above)
Comment (dtrifiro) [reply examples/example.py:6]: Let's get rid of these classes, and just use (optional) kwargs. We can do this in another PR

File: tests/test_api.py (hunk 6 above)
Comment (dtrifiro) [reply tests/test_api.py:1]: Some (naive) way of making sure that we keep `__all__` updated in `__init__.py`, we can eventually expand this to make sure that the API behaves as we expect it to

**Comment by dtrifiro** [comment]
@z103cb added a finalizer to close the channel. Also the channel is now a (private) attribute of the `GrpcClient` class

**Review by z103cb — APPROVED** [review]
LTGM>
